runtime/sema/type.go

```diff
@@ -3578,7 +3607,11 @@ func (t *CompositeType) GetMembers() map[string]MemberResolver {
 }
 
 func (t *CompositeType) IsResourceType() bool {
-	return t.Kind == common.CompositeKindResource
+	return t.Kind == common.CompositeKindResource ||
+		// attachments are always the same kind as their base type
+		(t.Kind == common.CompositeKindAttachment &&
+			t.baseType != t &&
```

Comment:
Why do we need to specifically look for this "self-reference"?

Reply:
`attachment A for A {}` will cause a stack overflow otherwise, as we will continuously call into `isResourceType` over and over again.

Reply:
I see! Could you please add this explanation and example as a comment to the code?